compiler/qsc_eval/src/lib.rs

```diff
@@ -400,17 +385,25 @@ impl<'a, G: GlobalLookup<'a>> State<'a, G> {
         self.vals.push(val);
     }
 
-    pub(crate) fn eval(&mut self) -> Result<Value, (Error, CallStack)> {
+    /// # Errors
+    /// Returns the first error encountered during execution.
+    pub fn eval(
+        &mut self,
+        globals: &impl GlobalLookup<'a>,
+        env: &mut Env,
+        sim: &mut impl Backend,
+        out: &mut impl Receiver,
```

Comment:
The combination of GlobalLookup, Env, Backend and Receiver is used a lot. Can we have a new struct for them?

Reply:
This requires more work and design. We can follow up in a another PR.